tools/gyp/pylib/gyp/__init__.py

```diff
@@ -104,8 +104,12 @@ def Load(
         default_variables.setdefault(key, val)
 
     output_dir = params["options"].generator_output or params["options"].toplevel_dir
-    default_variables.setdefault("PRODUCT_DIR_ABS", os.path.join(output_dir,
-        default_variables['CONFIGURATION_NAME']))
+    if (default_variables['GENERATOR'] == 'ninja'):
+      default_variables.setdefault("PRODUCT_DIR_ABS", os.path.join(output_dir,
+          'out', default_variables['build_type']))
+    else:
+      default_variables.setdefault("PRODUCT_DIR_ABS", os.path.join(output_dir,
+          default_variables['build_type']))
```

Comment:
@danbev should this sticks to `CONFIGURATION_NAME`, I run into some problems when directly run this files.